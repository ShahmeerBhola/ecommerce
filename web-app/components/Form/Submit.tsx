import React, { FC } from 'react';
import Button from '../Button';

type OwnProps = {
    isSubmitting: boolean;
    loadingMessage?: string;
};

type AllProps = OwnProps & React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>;

const Submit: FC<AllProps> = ({ disabled, loadingMessage = 'Loading', isSubmitting, children }) => {
    return (
        <Button
            type="submit"
            textSize="text-sm"
            disabled={disabled}
            className="w-full h-full"
            onMouseDown={(event): void => {
                // Fixes https://github.com/standout-specialties/ecommerce/issues/59
                // https://github.com/formium/formik/issues/1796
                event.preventDefault();
            }}
        >
            {isSubmitting ? loadingMessage : children}
        </Button>
    );
};

export default Submit;
