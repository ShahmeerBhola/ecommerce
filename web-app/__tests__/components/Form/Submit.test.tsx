import React from 'react';
import renderer from 'react-test-renderer';
import Submit from '../../../components/Form/Submit';

describe('Submit component', () => {
  test('it renders properly when isSubmitting is false', () => {
    const tree = renderer
      .create(
        <Submit isSubmitting={false}>
          <p>Hello</p>
        </Submit>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly default loadingMessage when isSubmitting is true', () => {
    const tree = renderer
      .create(
        <Submit isSubmitting={true}>
          <p>Hello</p>
        </Submit>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when isSubmitting is true and loadingMessage is provided', () => {
    const tree = renderer
      .create(
        <Submit isSubmitting={true} loadingMessage="Yoooo loading!">
          <p>Hello</p>
        </Submit>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when classNameProp is provided', () => {
    const tree = renderer
      .create(
        <Submit isSubmitting className="my-custom-class">
          <p>Hello</p>
        </Submit>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when disabled is false', () => {
    const tree = renderer
      .create(
        <Submit isSubmitting disabled={false}>
          <p>Hello</p>
        </Submit>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when disabled is true', () => {
    const tree = renderer
      .create(
        <Submit isSubmitting disabled={true}>
          <p>Hello</p>
        </Submit>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
