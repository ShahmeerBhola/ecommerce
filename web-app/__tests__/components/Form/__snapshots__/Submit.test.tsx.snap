// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit component it renders properly default loadingMessage when isSubmitting is true 1`] = `
<button
  className="font-secondary font-bold py-2 px-5 uppercase disabled:opacity-75 disabled:cursor-not-allowed focus:outline-none h-full w-full text-sm text-white bg-red-100 hover:bg-red-200"
  type="submit"
>
  Loading
</button>
`;

exports[`Submit component it renders properly when classNameProp is provided 1`] = `
<button
  className="font-secondary font-bold py-2 px-5 uppercase disabled:opacity-75 disabled:cursor-not-allowed focus:outline-none my-custom-class text-white bg-red-100 hover:bg-red-200"
  type="submit"
>
  Loading
</button>
`;

exports[`Submit component it renders properly when disabled is false 1`] = `
<button
  className="font-secondary font-bold py-2 px-5 uppercase disabled:opacity-75 disabled:cursor-not-allowed focus:outline-none h-full w-full text-sm text-white bg-red-100 hover:bg-red-200"
  disabled={false}
  type="submit"
>
  Loading
</button>
`;

exports[`Submit component it renders properly when disabled is true 1`] = `
<button
  className="font-secondary font-bold py-2 px-5 uppercase disabled:opacity-75 disabled:cursor-not-allowed focus:outline-none h-full w-full text-sm text-white bg-red-100 hover:bg-red-200"
  disabled={true}
  type="submit"
>
  Loading
</button>
`;

exports[`Submit component it renders properly when isSubmitting is false 1`] = `
<button
  className="font-secondary font-bold py-2 px-5 uppercase disabled:opacity-75 disabled:cursor-not-allowed focus:outline-none h-full w-full text-sm text-white bg-red-100 hover:bg-red-200"
  type="submit"
>
  <p>
    Hello
  </p>
</button>
`;

exports[`Submit component it renders properly when isSubmitting is true and loadingMessage is provided 1`] = `
<button
  className="font-secondary font-bold py-2 px-5 uppercase disabled:opacity-75 disabled:cursor-not-allowed focus:outline-none h-full w-full text-sm text-white bg-red-100 hover:bg-red-200"
  type="submit"
>
  Yoooo loading!
</button>
`;
